.circleci/config.yml

```diff
@@ -134,12 +137,19 @@ jobs:
             test/test_path_preservation.ps1
 
   build-docker-image:
-    executor: bionic
+    executor: ubuntu
     steps:
       - checkout
       - run:
           name: install docker
-          command: apt-get update -q && apt-get install -q -y docker.io
+          command: |
+            apt-get update -q 
+            apt-get install -q -y ca-certificates curl gnupg lsb-release
+            mkdir -p /etc/apt/keyrings
+            curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+            echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+            apt-get update -q
+            apt-get install -q -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
```

Comment:
Why is all this extra stuff needed here?

Reply:
Docker has updated the recommended installation method, see https://docs.docker.com/engine/install/ubuntu/#uninstall-old-versions
I think since they have updated the system version, these can be updated together.

Reply:
Is that needed as part of this change to a new ubuntu version?  If not, lets make that a separate change maybe?

Reply:
No, it is not necessary

Reply:
> Is that needed as part of this change to a new ubuntu version? If not, lets make that a separate change maybe?

However, as of now, I don't see the new versions 23.0.0 and 21.10.22/23 in the docker.io package, so I think it's best to use the new installation method

Reply:
Can you remove this part of the change and submit a separate PR?     Seems like this is switching from the ubuntu packaged docker to directly using the upstream pacakges.  We can discuss that separately there maybe?

Reply:
Okay, I'll revert this change first, and open a separate one later

Reply:
After switching back to the old way there was a problem with daemon not starting automatically. I tried ssh connection and then manually started docker with the following error:
```shell
root@9cdcd1906338: ~/projectroot@9cdcd1906338:~/project# /usr/bin/dockerd
INFO[2023-02-08T17:16:59.597232157Z] Starting up                                  
INFO[2023-02-08T17:16:59.598653570Z] libcontainerd: started new containerd process  pid=1044
INFO[2023-02-08T17:16:59.598990184Z] parsed scheme: "unix"                         module=grpc
INFO[2023-02-08T17:16:59.599003352Z] scheme "unix" not registered, fallback to default scheme  module=grpc
INFO[2023-02-08T17:16:59.599023135Z] ccResolverWrapper: sending update to cc: {[{unix:///var/run/docker/containerd/containerd.sock  <nil> 0 <nil>}] <nil> <nil>}  module=grpc
INFO[2023-02-08T17:16:59.599036101Z] ClientConn switching balancer to "pick_first"  module=grpc
WARN[0000] deprecated version : `1`, please switch to version `2` 
INFO[2023-02-08T17:16:59.637728889Z] starting containerd                           revision= version="1.5.9-0ubuntu1~20.04.6"
INFO[2023-02-08T17:16:59.657022988Z] loading plugin "io.containerd.content.v1.content"...  type=io.containerd.content.v1
INFO[2023-02-08T17:16:59.657089145Z] loading plugin "io.containerd.snapshotter.v1.aufs"...  type=io.containerd.snapshotter.v1
INFO[2023-02-08T17:16:59.657191573Z] loading plugin "io.containerd.snapshotter.v1.btrfs"...  type=io.containerd.snapshotter.v1
INFO[2023-02-08T17:16:59.657408969Z] skip loading plugin "io.containerd.snapshotter.v1.btrfs"...  error="path /var/lib/docker/containerd/daemon/io.containerd.snapshotter.v1.btrfs (overlay) must be a btrfs filesystem to be used with the btrfs snapshotter: skip plugin" type=io.containerd.snapshotter.v1
INFO[2023-02-08T17:16:59.657430926Z] loading plugin "io.containerd.snapshotter.v1.devmapper"...  type=io.containerd.snapshotter.v1
WARN[2023-02-08T17:16:59.657451965Z] failed to load plugin io.containerd.snapshotter.v1.devmapper  error="devmapper not configured"
INFO[2023-02-08T17:16:59.657463826Z] loading plugin "io.containerd.snapshotter.v1.native"...  type=io.containerd.snapshotter.v1
INFO[2023-02-08T17:16:59.657486806Z] loading plugin "io.containerd.snapshotter.v1.overlayfs"...  type=io.containerd.snapshotter.v1
INFO[2023-02-08T17:16:59.658049391Z] loading plugin "io.containerd.snapshotter.v1.zfs"...  type=io.containerd.snapshotter.v1
INFO[2023-02-08T17:16:59.658252751Z] skip loading plugin "io.containerd.snapshotter.v1.zfs"...  error="path /var/lib/docker/containerd/daemon/io.containerd.snapshotter.v1.zfs must be a zfs filesystem to be used with the zfs snapshotter: skip plugin" type=io.containerd.snapshotter.v1
INFO[2023-02-08T17:16:59.658272166Z] loading plugin "io.containerd.metadata.v1.bolt"...  type=io.containerd.metadata.v1
WARN[2023-02-08T17:16:59.658291790Z] could not use snapshotter devmapper in metadata plugin  error="devmapper not configured"
INFO[2023-02-08T17:16:59.658302759Z] metadata content store policy set             policy=shared
INFO[2023-02-08T17:16:59.658510986Z] loading plugin "io.containerd.differ.v1.walking"...  type=io.containerd.differ.v1
INFO[2023-02-08T17:16:59.658529707Z] loading plugin "io.containerd.gc.v1.scheduler"...  type=io.containerd.gc.v1
INFO[2023-02-08T17:16:59.658562722Z] loading plugin "io.containerd.service.v1.introspection-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.658590386Z] loading plugin "io.containerd.service.v1.containers-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.658604298Z] loading plugin "io.containerd.service.v1.content-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.658617846Z] loading plugin "io.containerd.service.v1.diff-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.658632425Z] loading plugin "io.containerd.service.v1.images-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.658646337Z] loading plugin "io.containerd.service.v1.leases-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.658660716Z] loading plugin "io.containerd.service.v1.namespaces-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.658674809Z] loading plugin "io.containerd.service.v1.snapshots-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.658688461Z] loading plugin "io.containerd.runtime.v1.linux"...  type=io.containerd.runtime.v1
INFO[2023-02-08T17:16:59.658988183Z] loading plugin "io.containerd.runtime.v2.task"...  type=io.containerd.runtime.v2
INFO[2023-02-08T17:16:59.659048161Z] loading plugin "io.containerd.monitor.v1.cgroups"...  type=io.containerd.monitor.v1
INFO[2023-02-08T17:16:59.659892962Z] loading plugin "io.containerd.service.v1.tasks-service"...  type=io.containerd.service.v1
INFO[2023-02-08T17:16:59.659937347Z] loading plugin "io.containerd.internal.v1.restart"...  type=io.containerd.internal.v1
INFO[2023-02-08T17:16:59.659985473Z] loading plugin "io.containerd.grpc.v1.containers"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660140409Z] loading plugin "io.containerd.grpc.v1.content"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660160719Z] loading plugin "io.containerd.grpc.v1.diff"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660177443Z] loading plugin "io.containerd.grpc.v1.events"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660190131Z] loading plugin "io.containerd.grpc.v1.healthcheck"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660207561Z] loading plugin "io.containerd.grpc.v1.images"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660221505Z] loading plugin "io.containerd.grpc.v1.leases"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660233711Z] loading plugin "io.containerd.grpc.v1.namespaces"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660246460Z] loading plugin "io.containerd.internal.v1.opt"...  type=io.containerd.internal.v1
INFO[2023-02-08T17:16:59.660283010Z] loading plugin "io.containerd.grpc.v1.snapshots"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660298546Z] loading plugin "io.containerd.grpc.v1.tasks"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660311877Z] loading plugin "io.containerd.grpc.v1.version"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660325599Z] loading plugin "io.containerd.grpc.v1.introspection"...  type=io.containerd.grpc.v1
INFO[2023-02-08T17:16:59.660509024Z] serving...                                    address=/var/run/docker/containerd/containerd-debug.sock
INFO[2023-02-08T17:16:59.660580951Z] serving...                                    address=/var/run/docker/containerd/containerd.sock.ttrpc
INFO[2023-02-08T17:16:59.660639047Z] serving...                                    address=/var/run/docker/containerd/containerd.sock
INFO[2023-02-08T17:16:59.660657130Z] containerd successfully booted in 0.023790s  
INFO[2023-02-08T17:16:59.664656375Z] parsed scheme: "unix"                         module=grpc
INFO[2023-02-08T17:16:59.664689485Z] scheme "unix" not registered, fallback to default scheme  module=grpc
INFO[2023-02-08T17:16:59.664705798Z] ccResolverWrapper: sending update to cc: {[{unix:///var/run/docker/containerd/containerd.sock  <nil> 0 <nil>}] <nil> <nil>}  module=grpc
INFO[2023-02-08T17:16:59.664716391Z] ClientConn switching balancer to "pick_first"  module=grpc
INFO[2023-02-08T17:16:59.665349989Z] parsed scheme: "unix"                         module=grpc
INFO[2023-02-08T17:16:59.665371205Z] scheme "unix" not registered, fallback to default scheme  module=grpc
INFO[2023-02-08T17:16:59.665386969Z] ccResolverWrapper: sending update to cc: {[{unix:///var/run/docker/containerd/containerd.sock  <nil> 0 <nil>}] <nil> <nil>}  module=grpc
INFO[2023-02-08T17:16:59.665397481Z] ClientConn switching balancer to "pick_first"  module=grpc
ERRO[2023-02-08T17:16:59.666471002Z] failed to mount overlay: operation not permitted  storage-driver=overlay2
ERRO[2023-02-08T17:16:59.666528603Z] exec: "fuse-overlayfs": executable file not found in $PATH  storage-driver=fuse-overlayfs
ERRO[2023-02-08T17:16:59.666600428Z] AUFS cannot be used in non-init user namespace  storage-driver=aufs
ERRO[2023-02-08T17:16:59.666984325Z] failed to mount overlay: operation not permitted  storage-driver=overlay
WARN[2023-02-08T17:16:59.669131130Z] Your kernel does not support CPU realtime scheduler 
WARN[2023-02-08T17:16:59.669165030Z] Your kernel does not support cgroup blkio weight 
WARN[2023-02-08T17:16:59.669171719Z] Your kernel does not support cgroup blkio weight_device 
INFO[2023-02-08T17:16:59.669321203Z] Loading containers: start.                   
WARN[2023-02-08T17:16:59.670720394Z] Running iptables --wait -t nat -L -n failed with message: `iptables v1.8.4 (legacy): can't initialize iptables table `nat': Permission denied (you must be root)
Perhaps iptables or your kernel needs to be upgraded.`, error: exit status 3 
INFO[2023-02-08T17:16:59.687118997Z] stopping event stream following graceful shutdown  error="<nil>" module=libcontainerd namespace=moby
INFO[2023-02-08T17:16:59.687435414Z] stopping healthcheck following graceful shutdown  module=libcontainerd
INFO[2023-02-08T17:16:59.687450875Z] stopping event stream following graceful shutdown  error="context canceled" module=libcontainerd namespace=plugins.moby
WARN[2023-02-08T17:17:00.688281130Z] grpc: addrConn.createTransport failed to connect to {unix:///var/run/docker/containerd/containerd.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix:///var/run/docker/containerd/containerd.sock: timeout". Reconnecting...  module=grpc
failed to start daemon: Error initializing network controller: error obtaining controller instance: failed to create NAT chain DOCKER: iptables failed: iptables -t nat -N DOCKER: iptables v1.8.4 (legacy): can't initialize iptables table `nat': Permission denied (you must be root)
Perhaps iptables or your kernel needs to be upgraded.
```

Reply:
So I think there is still Install it in a new way

Reply:
Reverted to the new installation